src/Storage/SqlContentEntityStorageTrait.php

```diff
@@ -121,8 +121,24 @@ protected function mapFromStorageRecords(array $records, $load_from_revision = F
    * {@inheritdoc}
    */
   protected function postLoad(array &$entities) {
-    $entity_class = $this->getEntityClass();
-    $entity_class::postLoad($this, $entities);
+    // Call postLoad() by bundle if bundleKey is set.
+    // Iterate over entities to determine respective bundles because entities
+    // might are of different types.
+    if (isset($this->bundleKey)) {
+      $entities_by_bundle = [];
+      foreach ($entities as $entity) {
+        $entities_by_bundle[$entity->get($this->bundleKey)->getString()][] = $entity;
```

Comment:
@simonbaese Could we not use something like just `$entity->bundle()`?

Reply:
@amcgowanca I was not really sure because you were checking for the `bundleKey` throughout the code. The method `bundle()` specifically looks for the value by the key `'bundle'` in the entity (see `ContentEntityBase`) and as it is public it could also be overwritten.